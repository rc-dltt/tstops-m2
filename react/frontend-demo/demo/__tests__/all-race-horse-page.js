import React from "react";
import { MockedProvider } from '@apollo/client/testing';
import { PaperProvider } from 'react-native-paper';
import AllRaceHorsePage from "../Components/AllRaceHorsePage";
import { allRaceQuery, allHorseQuery } from "../graphql/query"
import { loginMutation } from '../graphql/mutation';
import { render } from "@testing-library/react-native";

const mocks = [
    {
        request: {
            query: loginMutation,
            variables: {
                email: 'johndoe@email.com',
                password: 'pAsSWoRd!',
            },
        },
        result: {
            data: {
                login: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6InVzZXIjMSIsInJvbGVzIjpbInVzZXIiLCJhZG1pbiJdLCJpYXQiOjE2OTI2NzE5NTEsImV4cCI6MTY5Mjc1ODM1MSwic3ViIjoidXNlciMxIn0.3CPT3fQqU6fLarIh8y0zFkJD7Xik5UdzBBkapqZdzSM"
            },
        },
    },
    {
        request: {
            query: allRaceQuery,
        },
        result: {
            data: {
                races: {
                    id: '84',
                    no: "#1",
                    startTime: "2023-08-12",
                    venue: "Shat Tin"
                }
            },
        },
    },
    {
        request: {
            query: allHorseQuery,
        },
        result: {
            data: {
                horses: {
                    id: '103',
                    name: "Test Horse",
                    rank: "123"
                }
            },
        },
    },
];

beforeEach(() => {
    jest.useFakeTimers();
});

describe('All Race / Horse Page', () => {
    it('Snapshot Test', () => {
        const allRaceHorsePage = render(
            <MockedProvider mocks={mocks} addTypename={false} >
                <PaperProvider>
                    <AllRaceHorsePage />
                </PaperProvider>
            </MockedProvider>
        );
        expect(allRaceHorsePage).toMatchSnapshot();
    })
})
