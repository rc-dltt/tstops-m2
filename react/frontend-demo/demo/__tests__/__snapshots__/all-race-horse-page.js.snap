// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Race / Horse Page Snapshot Test 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      contentInsetAdjustmentBehavior="automatic"
    >
      <View>
        <View
          style={
            {
              "backgroundColor": "#FFF",
            }
          }
        >
          <View
            style={
              {
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "fontSize": 24,
                    "fontWeight": "600",
                  },
                  {
                    "color": "#000",
                  },
                ]
              }
            >
              Query - All Races
            </Text>
            <Text
              style={
                [
                  {
                    "fontSize": 18,
                    "fontWeight": "400",
                    "marginTop": 8,
                  },
                  {
                    "color": "#444",
                  },
                ]
              }
            >
              Details of all races.
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "#FFF",
            }
          }
        >
          <View
            style={
              {
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "fontSize": 24,
                    "fontWeight": "600",
                  },
                  {
                    "color": "#000",
                  },
                ]
              }
            >
              Query - All Horses
            </Text>
            <Text
              style={
                [
                  {
                    "fontSize": 18,
                    "fontWeight": "400",
                    "marginTop": 8,
                  },
                  {
                    "color": "#444",
                  },
                ]
              }
            >
              Details of all horses.
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaProvider>
`;
